xbmc/pvr/PVRManager.cpp

```diff
@@ -594,12 +573,31 @@ bool CPVRManager::SetWakeupCommand(void)
 
 void CPVRManager::OnSleep()
 {
+  SetWakeupCommand();
+  SaveLastPlayedChannel();
+
   g_PVRClients->OnSystemSleep();
 }
 
 void CPVRManager::OnWake()
 {
   g_PVRClients->OnSystemWake();
+
+  if (IsStarted())
```

Comment:
IMO every method should be self-contained. They need to validate the state if necessary. We should drop that check in OnWake.